### ISSUE TITLE
[Bug] NullReferenceException in MaterialActivityIndicatorRenderer

### ISSUE THREAD

**Issue by twenzel** [xamarin/Xamarin.Forms#6968]
### Description
I can reproduce the bug on my side, but can't tell you what is going on. ;(

  at Xamarin.Forms.Material.iOS.MaterialActivityIndicatorRenderer.LayoutSubviews () [0x00006] in D:\a\1\s\Xamarin.Forms.Material.iOS\MaterialActivityIndicatorRenderer.cs:82 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend(intptr,intptr)
  at CoreAnimation.CALayer.LayoutSublayers () [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/CoreAnimation/CALayer.g.cs:490 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker+<>c__DisplayClass28_0.<OnUpdateNativeControl>g__update|0 () [0x00129] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:247 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.OnUpdateNativeControl (CoreAnimation.CALayer caLayer) [0x002be] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:303 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.UpdateNativeControl () [0x0004f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:353 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00055] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:331 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker..ctor (Xamarin.Forms.Platform.iOS.IVisualElementRenderer renderer, System.Boolean trackFrame) [0x0006b] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:48 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker..ctor (Xamarin.Forms.Platform.iOS.IVisualElementRenderer renderer) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:34 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x0009f] in <6d6f830f76b442a8953f9ae13539c687>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00000] in <6d6f830f76b442a8953f9ae13539c687>:0 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:182 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.OnChildAdded (Xamarin.Forms.VisualElement view) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:97 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.Load () [0x0001e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:46 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000de] in <6d6f830f76b442a8953f9ae13539c687>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00000] in <6d6f830f76b442a8953f9ae13539c687>:0 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:182 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.OnChildAdded (Xamarin.Forms.VisualElement view) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:97 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.Load () [0x0001e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:46 
  at Xamarin.Forms.Platform.iOS.PageRenderer.ViewDidLoad () [0x0008f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:221 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.IntPtr_objc_msgSendSuper(intptr,intptr)
  at UIKit.UIViewController.get_View () [0x0002a] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/UIKit/UIViewController.g.cs:3037 
  at Xamarin.Forms.Platform.iOS.PageRenderer.get_NativeView () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:97 
  at Xamarin.Forms.Platform.iOS.PageRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:113 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:182 
  at Xamarin.Forms.Platform.iOS.NavigationRenderer.CreateViewControllerForPage (Xamarin.Forms.Page page) [0x00008] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\NavigationRenderer.cs:358 
  at Xamarin.Forms.Platform.iOS.NavigationRenderer.OnPushAsync (Xamarin.Forms.Page page, System.Boolean animated) [0x0001d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\NavigationRenderer.cs:345 
  at Xamarin.Forms.Platform.iOS.NavigationRenderer.<ViewDidLoad>b__46_0 (Xamarin.Forms.Page p) [0x00024] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\NavigationRenderer.cs:238 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1023 
  at Foundation.NSAsyncSynchronizationContextDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/Foundation/NSAction.cs:178 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/UIKit/UIApplication.cs:86 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.10.0.157/src/Xamarin.iOS/UIKit/UIApplication.cs:65 
  at eHealth.iOS.Application.Main (System.String[] args) [0x00001] in F:\E-Health\E-Health\src\eHealth.iOS\Main.cs:17 

### Versions
Xamarin.Forms: 4.1.0.555618
Xamarin.Forms.Visual.Material: 4.1.0.555618

**Comment by PureWeen** [comment]
@twenzel can you provide a sample ?

The exception is coming from this line
```
public override void LayoutSubviews()
		{
			base.LayoutSubviews();

			// try get the radius for this size
			var min = NMath.Min(Control.Bounds.Width, Control.Bounds.Height);
```

So it looks like it's trying to layout an already disposed of ActivityIndicator. Are you reusing the Activity Indicator or reusing a page? Or reparenting something? Moving a control around?

**Comment by samhouts** [comment]
@twenzel  Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!

**Comment by ivoryguard** [comment]
I also have experienced the same issue.
My app has worked well with XF 4.1 and iPhone 5.
However, when I tested on iPAD 6gen recently, my app was crashed right now after a app shell content page was re-visited from another page.
The only change was updating Xamarin forms from 4.1 to 4.2 latest version.
The error message is as this;



```
-		$exception	{System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.Material.iOS.MaterialActivityIndicatorRenderer.LayoutSubviews () [0x00006] in D:\a\1\s\Xamarin.Forms.Material.iOS\MaterialActivityIndicatorRenderer.cs:82 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.void_objc_msgSend(intptr,intptr)
  at CoreAnimation.CALayer.LayoutSublayers () [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.114/src/Xamarin.iOS/CoreAnimation/CALayer.g.cs:490 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker+<>c__DisplayClass28_0.<OnUpdateNativeControl>g__update|0 () [0x00129] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:247 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.OnUpdateNativeControl (CoreAnimation.CALayer caLayer) [0x002be] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:303 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.UpdateNativeControl () [0x0004f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:353 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker.SetElement (Xamarin.Forms.VisualElement oldElement, Xamarin.Forms.VisualElement newElement) [0x00055] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:331 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker..ctor (Xamarin.Forms.Platform.iOS.IVisualElementRenderer renderer, System.Boolean trackFrame) [0x0006b] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:48 
  at Xamarin.Forms.Platform.iOS.VisualElementTracker..ctor (Xamarin.Forms.Platform.iOS.IVisualElementRenderer renderer) [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementTracker.cs:34 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x0009f] in <f0489d3941b24a20a5b9223b572ec7c0>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00000] in <f0489d3941b24a20a5b9223b572ec7c0>:0 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:188 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.OnChildAdded (Xamarin.Forms.VisualElement view) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:97 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.Load () [0x0001e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:46 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].SetElement (TElement element) [0x000de] in <f0489d3941b24a20a5b9223b572ec7c0>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].Xamarin.Forms.Platform.iOS.IVisualElementRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x00000] in <f0489d3941b24a20a5b9223b572ec7c0>:0 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:188 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.OnChildAdded (Xamarin.Forms.VisualElement view) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:97 
  at Xamarin.Forms.Platform.iOS.VisualElementPackager.Load () [0x0001e] in D:\a\1\s\Xamarin.Forms.Platform.iOS\VisualElementPackager.cs:46 
  at Xamarin.Forms.Platform.iOS.PageRenderer.ViewDidLoad () [0x0008f] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:235 
  at (wrapper managed-to-native) ObjCRuntime.Messaging.IntPtr_objc_msgSendSuper(intptr,intptr)
  at UIKit.UIViewController.get_View () [0x0002a] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.114/src/Xamarin.iOS/UIKit/UIViewController.g.cs:3037 
  at Xamarin.Forms.Platform.iOS.PageRenderer.get_NativeView () [0x00000] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:101 
  at Xamarin.Forms.Platform.iOS.PageRenderer.SetElement (Xamarin.Forms.VisualElement element) [0x0003d] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\PageRenderer.cs:117 
  at Xamarin.Forms.Platform.iOS.Platform.CreateRenderer (Xamarin.Forms.VisualElement element) [0x00014] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Platform.cs:188 
  at Xamarin.Forms.Platform.iOS.ShellSectionRootRenderer.LoadRenderers () [0x00028] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\ShellSectionRootRenderer.cs:146 
  at Xamarin.Forms.Platform.iOS.ShellSectionRootRenderer.ViewDidLoad () [0x00035] in D:\a\1\s\Xamarin.Forms.Platform.iOS\Renderers\ShellSectionRootRenderer.cs:63 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.114/src/Xamarin.iOS/UIKit/UIApplication.cs:86 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.114/src/Xamarin.iOS/UIKit/UIApplication.cs:65 
  at MyTestApp.iOS.Application.Main (System.String[] args) [0x00001] in D:\MySource\MyTestApp\MyTestApp.iOS\Main.cs:17 }	System.NullReferenceException
-		base	{System.SystemException}	System.SystemException
-		base	{System.Exception}	System.Exception
+		Data	{System.Collections.ListDictionaryInternal}	System.Collections.ListDictionaryInternal
		HResult	-2147467261	int
		HelpLink	(null)	string
		InnerException	(null)	System.Exception
		Message	"Object reference not set to an instance of an object"	string
		Source	"Xamarin.Forms.Material"	string
		StackTrace	"  at Xamarin.Forms.Material.iOS.MaterialActivityIndicatorRenderer.LayoutSubviews () [0x00006] in D:\\…"	string
+		TargetSite	{System.Reflection.RuntimeMethodInfo}	System.Reflection.RuntimeMethodInfo
+		Static members		
+		Non-public members		

```


This error must be from Xamarin Forms Materials.
After I deleted the following line at XAML, it has gone.

```
Visual="Material"
```


Environment:
Visual Studio 2019 16.2.3
Xamarin Forms 4.2.0.709249

**Comment by caesarm** [comment]
I recreated this bug using the MasterDetail template. 

To trigger it, wait for the Items page to load and the activity indicator to disappear.
Use the menu to open the About page. 
The exception will occur when you browse back to the ItemsPage.

I noticed that the IsVisible seems to be critical in causing the issue. Commenting it out below doesn't crash.

            ActivityIndicator.IsVisible = true;
            ActivityIndicator.IsRunning = true;

            await Task.Delay(2000);

            ActivityIndicator.IsVisible = false;
            ActivityIndicator.IsRunning = false;


[MaterialActivityIndicatorRenderer.zip](https://github.com/xamarin/Xamarin.Forms/files/3860424/MaterialActivityIndicatorRenderer.zip)
